pkg/blockbuilder/scheduler/scheduler.go

```diff
@@ -155,17 +155,44 @@ func (s *BlockScheduler) HandleSyncJob(_ context.Context, builderID string, job
 	return nil
 }
 
-// unimplementedScheduler provides default implementations that panic.
-type unimplementedScheduler struct{}
-
-func (s unimplementedScheduler) HandleGetJob(_ context.Context, _ string) (*types.Job, bool, error) {
-	panic("unimplemented")
+func (s *BlockScheduler) CompleteJob(_ context.Context, req *proto.CompleteJobRequest) (*proto.CompleteJobResponse, error) {
+	s.queue.MarkComplete(req.Job.Id)
+	return nil, nil
 }
 
-func (s unimplementedScheduler) HandleCompleteJob(_ context.Context, _ string, _ *types.Job) error {
-	panic("unimplemented")
+func (s *BlockScheduler) SyncJob(_ context.Context, req *proto.SyncJobRequest) (*proto.SyncJobResponse, error) {
+	s.queue.SyncJob(req.Job.Id, req.BuilderId, &types.Job{
+		ID:        req.Job.Id,
+		Partition: req.Job.Partition,
+		Offsets: types.Offsets{
+			Min: req.Job.Offsets.Min,
+			Max: req.Job.Offsets.Max,
+		},
+	})
+
+	return nil, nil
```

Comment:
we should return non-nil response types to satisfy caller expectations even though they're basically empty structs.